src/main/java/org/rulez/demokracia/pdengine/SessionFactoryManager.java

```diff
@@ -26,5 +30,13 @@ public String getWsUserName() {
 	public boolean hasAssurance(final String role) {
 		return getWsContext().isUserInRole(role);
 	}
+	
+	public FakeADAAssuranceProvider getFakeADAAssuranceProvider() {
```

Comment:
Have an interface ADAAssuranceProvider in the production code.
Make FakeADAAssuranceProvider implement that interface.
for now have an
`ADAAssuranceProvider assuranceProvider;`
field in this class.
In the test code just assign an instance of FakeADAAssuranceProvider to the field.
https://medium.freecodecamp.org/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f